### ISSUE TITLE
Whether soil can be used as a target for detection

### ISSUE THREAD

**Issue by chenfh21** [PRBonn/phenobench-baselines#9]
[]([](https://github.com/PRBonn/phenobench-baselines/blob/78db625441e54c5b64efabeb0d886d020961665b/plant_detection/yolov7/src/data/phenobench_plants.yaml#L10))
Hi:
    I'm curious about soil as an object in the Phenobench plant object detection task. I checked the objects in the MSCOCO dataset (80 & 91), and I understand the soil can be concluded as the 'stuff' category as it is without clear boundaries. I'm confused about how Phenobench determines it as an individual object.  And whether transfer learning was used during training (I carefully read your latest paper but couldn't find any information on whether pre-trained weights were used in the 300 iterations).

  In addition, I use the YOLOV7 default settings when conducting experiments on the network I proposed, and all other indicators performed well except for the increase in val/obj-loss. I would like to know if you have also encountered such a problem. The reason given by the community is the inconsistency caused by using computeLossOTA during training and computeLoss during validation (ref: https://github.com/WongKinYiu/yolov7/issues/326). 
![111](https://github.com/user-attachments/assets/87dfb170-5c19-4f5b-948d-e5109465befe)

**Comment by jbehley** [comment]
Also, here, sorry for not following up earlier: I personally haven't trained the YOLO baseline and therefore cannot comment on the behavior.

Maybe @yuelinn can comment on this, as she trained the YOLO baselines.

**Comment by yuelinn** [comment]
Hi, we did not train YOLO with soil as a class, only plants (crops and weeds class) and crop leaves for the two tasks respectively. You could try to train with soil class yourself, we used the code from the dev kit to train our YOLO, maybe you can try to use that code but with your own soil bounding boxes which can be easily generated from the semantic segmentation network. 

Sorry, I do not remember what the object loss curve was like for us. As mentioned, you can try to check by rerunning the YOLO code as is; but I do recall sometimes the object loss curve is not quite "normal" but this doesn't necessarily indicate a lower mAP in the end. If you like to get a nice curve, maybe you could tune the weights of the losses.

**Comment by chenfh21** [comment]
> Hi, we did not train YOLO with soil as a class, only plants (crops and weeds class) and crop leaves for the two tasks respectively. You could try to train with soil class yourself, we used the code from the dev kit to train our YOLO, maybe you can try to use that code but with your own soil bounding boxes which can be easily generated from the semantic segmentation network.
> 
> Sorry, I do not remember what the object loss curve was like for us. As mentioned, you can try to check by rerunning the YOLO code as is; but I do recall sometimes the object loss curve is not quite "normal" but this doesn't necessarily indicate a lower mAP in the end. If you like to get a nice curve, maybe you could tune the weights of the losses.

I use the data configuration script you released, and I also try to remove 'soil', but ran an error. Of course, according to this configuration, there is no indicator information about 'soil' in the final presentation results, namely the curve and validation results. I am just surprised that it is listed as a category in the configuration file, and I suspect that it is because the data conversion preserves the information of that category.
![222](https://github.com/user-attachments/assets/246b750d-db6d-41ef-b2c7-9388566f08f3)

**Comment by yuelinn** [comment]
Oh, I see where the confusion is. I left the label for soil only to keep the class ids consistent with the semantic segmentation task. This is perhaps a slightly confusing approach to acheive this but it should not affect the training. As you said, there are no soil labels used and no soil output in the detection. But the weights we publish were trained for 3 classes so it so you need to keep the configuration as is if you want to use our weights. You could also train your own weights without the soil class.

-------- Original Message --------

**Comment by chenfh21** [comment]
https://github.com/PRBonn/phenobench-baselines/issues/9#issuecomment-2428677032
Thank you very much for your reply!

**Comment by chenfh21** [comment]
Another question，have pre-trained weights or transfer learning techniques been used in the target detection baseline？

**Comment by yuelinn** [comment]
from what I remember, I trained YOLO from scratch

**Comment by chenfh21** [comment]
> from what I remember, I trained YOLO from scratch
Thank you for your reply!
